x-pack/plugins/maps/server/routes.js

```diff
@@ -587,6 +597,47 @@ export function initRoutes(router, getLicenseId, emsSettings, kbnVersion, logger
     }
   );
 
+  if (drawingFeatureEnabled) {
+    router.post(
+      {
+        path: `/${INDEX_SOURCE_API_PATH}`,
+        validate: {
+          body: schema.object({
+            index: schema.string(),
+            mappings: schema.any(),
+          }),
+        },
+        options: {
+          body: {
+            accepts: ['application/json'],
+          },
+        },
+      },
+      async (context, request, response) => {
+        const { index, mappings } = request.body;
+        const indexPatternsService = await dataPlugin.indexPatterns.indexPatternsServiceFactory(
+          context.core.savedObjects.client,
+          context.core.elasticsearch.client.asCurrentUser
+        );
```

Comment:
Agreed that we need to verify user permissions. I previously chatted with @nreese about this and we agreed we should probably leverage the same solution that file upload ends up using to resolve https://github.com/elastic/kibana/issues/41928. This might be in the form of a dedicated endpoint to check permissions, not sure. Regardless I think we probably want to handle it separately from this PR since it's not straightforward yet how best to do this.